scripts/us_census/acs5yr/subject_tables/common/data_loader.py

```diff
@@ -0,0 +1,467 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common data processing module for ACS Subject Tables"""
+import os
+import sys
+import re
+import json
+from zipfile import ZipFile
+
+import numpy as np
+import pandas as pd
+
+# Allows the following module imports to work when running as a script
+_SCRIPT_PATH = os.path.dirname(os.path.abspath(__file__))
+
+sys.path.append(os.path.join(_SCRIPT_PATH, '.'))  # for resolve_geo_id
+from resolve_geo_id import convert_to_place_dcid
+
+_IGNORED_VALUES = set(['-', '*', '**', '***', '*****', 'N', '(X)', 'null'])
+
+
+def process_subject_tables(table_prefix='',
+                           input_path='./',
+                           output_dir='./',
+                           column_map_path=None,
+                           spec_path=None,
+                           debug=False,
+                           delimiter='!!',
+                           has_percent=False):
+    """
+    Wrapper method for invoking the data processing module, maps files with the
+    corresponding processing function based on input format (*.zip/*.csv),
+    output files are saved in the specified output directory.
+
+    NOTE: This module does not generate the column_map, you will need to run the
+  `generate_col_map.py` module before running this module in standalone mode.
+
+  NOTE: If the path for column_map is not specified this module will exit with
+  an error, stating that the column_map is a required input.
+
+  Args:
+  table_prefix: sting prefix that will be added to differentiate outputs
+  input_path: path to the input file(s). The module supports inputs as a either
+    a zip file or a csv file or a directory with zip/ csv files
+  output_dir: path to the output directory
+  column_map_path: path to the column_map generated with `generate_col_map.py`
+  spec_path: path to the JSON specification for the subject table
+  debug: if set, adds additional columns to output csv for debugging (default: False)
+  col_delimiter: String denoting the delimiter character in dataset (default: '!!')
```

Comment:
please use the same name as in the function signature: s/col_delimiter/delimiter/